src/core_io.h

```diff
@@ -25,7 +25,7 @@ std::string ScriptToAsmStr(const CScript& script, const bool fAttemptSighashDeco
 bool DecodeHexTx(CMutableTransaction& tx, const std::string& hex_tx, bool try_no_witness = false, bool try_witness = true);
 bool DecodeHexBlk(CBlock&, const std::string& strHexBlk);
 bool DecodeHexBlockHeader(CBlockHeader&, const std::string& hex_header);
-uint256 ParseHashStr(const std::string&, const std::string& strName);
+bool ParseHashStr(const std::string& strHex, uint256& result);
```

Comment:
Perhaps we can document `bool ParseHashStr(const std::string& strHex, uint256& result)` and specify the pre-condition that the `strHex` argument must be a hexadecimal string with a length of 64 bytes.